lib/jekyll-admin/path_helper.rb

```diff
@@ -51,6 +51,13 @@ def renamed?
       ensure_leading_slash(request_payload["path"]) != relative_path
     end
 
+    # Is this request creating a new file?
+    def new?
+      return false if request_payload["path"]
+
+      ensure_leading_slash(request_payload["path"]) != relative_path
+    end
+
```

Comment:
I'm a bit *unsure* about the duplicated logic in `:new?` and `:renamed?`..

Both methods ultimately test `ensure_leading_slash(request_payload["path"]) != relative_path` explicitly. But in the server namespaces, you have `ensure_not_written_file if new? || renamed?`.
So in `:new?` we test if `request_payload["path"]` is a truthy, and if so immediately call `:renamed?` method where `request_payload["path"]` is again tested for a truthy and then proceed if so..

Needs a bit of a DRYing up and refactoring, IMO.

Reply:
I have replaced

```
def new?
  return false if request_payload["path"]

  ensure_leading_slash(request_payload["path"]) != relative_path
end
```
```
def new?
  !request_payload["path"]
end
```

As `new?` doesn't need to compare the value, it only checks that it exists. So this becomes DRY for free.